schedule_view/schedule_view.py

```diff
@@ -168,25 +168,29 @@ def get_page_entries(start_time):
         mp_comment = get_mp_comment(week, mp_dat)
         if mp_comment is not None:
             entry["mp_comment"] = mp_comment
-        rltt_match = (cmds["source"] == week) & (
-            cmds["params"] == {"event_type": "RUNNING_LOAD_TERMINATION_TIME"}
-        )
-        if any(rltt_match):
-            rltt = cmds["date"][rltt_match][0]
-            entry["rltt"] = rltt
-            entry["date"] = rltt
-        else:
+        cmds_week = cmds[cmds["source"] == week]
+        rltt_cmd = cmds_week.get_rltt_cmd()
+        if rltt_cmd is None:
+            # This should just happen in the case when the start time is after rltt
+            # for the first "source" in run_loads
             continue
-        ss_match = (cmds["source"] == week) & (
-            cmds["params"] == {"event_type": "SCHEDULED_STOP_TIME"}
-        )
-        if any(ss_match):
-            ss = cmds["date"][ss_match][0]
-            entry["sched_stop"] = ss
+        entry["rltt"] = rltt_cmd["date"]
+        entry["date"] = rltt_cmd["date"]
+        sst_cmd = cmds_week.get_scheduled_stop_time_cmd()
+        if sst_cmd is not None:
+            # the date of the schedule stop time is the later of the sst date or
+            # the original sst
```

Comment:
This comment is not quite right since there is no comparison of the two to determine "later".

`scheduled_stop_time_orig` exists as a param if and only if the schedule was interrupted and it is guaranteed to be >= the command date. In theory your code logic could use this to uniquely determine a load interrupt instead of the more-complex logic of looking for extra load event commands within the interval. But what you are doing appears to be working so no need to change.


```suggestion
            # Date of the schedule stop time is either the original SST date (defined if and
            # only if the schedule was interrupted) or the SST command date.
```

Reply:
" In theory your code logic could use this to uniquely determine a load interrupt instead of the more-complex logic of looking for extra load event commands within the interval." - but only if we or this code remade the archived commands outside of the commands update interval.

Reply:
And I find command events useful on their own.